vast/vast.cpp

```diff
@@ -50,18 +50,31 @@ int main(int argc, char** argv) {
   }
   // Load plugins.
   auto loaded_plugin_paths = std::vector<std::filesystem::path>{};
-  auto plugin_files
+  auto plugin_files_or_names
     = caf::get_or(cfg, "vast.plugins", std::vector<std::string>{});
 #ifdef VAST_ENABLED_PLUGINS
   // If plugins are enabled at compile time to always be loaded, add them here
   // if they were not already configured as part of vast.plugins.
-  for (auto&& plugin_file : std::vector<std::string>{VAST_ENABLED_PLUGINS})
-    if (std::none_of(plugin_files.begin(), plugin_files.end(),
-                     [&](auto&& x) { return x == plugin_file; }))
-      plugin_files.push_back(std::move(plugin_file));
+  for (auto&& plugin_names : std::vector<std::string>{VAST_ENABLED_PLUGINS})
+    if (std::none_of(plugin_files_or_names.begin(), plugin_files_or_names.end(),
+                     [&](auto&& x) { return x == plugin_names; }))
+      plugin_files_or_names.push_back(std::move(plugin_names));
 #endif
   auto& plugins = plugins::get();
-  for (const auto& plugin_file : plugin_files) {
+  // Check if any of the specified plugins is already loaded as a static
+  // plugin, and silently remove it from the list of specified plugins if
+  // that's the case.
```

Comment:
```suggestion
  // that's the case.
  // TODO: Consider sorting the inputs and using `std::set_difference` instead
  // of this custom loop.
```

Reply:
Can do, but I'm not sure it's worth it over the trivial implementation. This is far from a critical piece of code and I doubt there'll ever be a large number of plugins loaded.

Reply:
I pushed it, but this is really damn ugly. Please take a look.

Reply:
Note that we have `detail/set_operations.hpp` that make these operations much easier to use.

Reply:
Nothing in here helps for this specific use case, especially because of the custom comparator.

Reply:
It seems this got a little out of hand while I was away. I originally asked to include a comment with the intent to inform the reader at a glance of what this statement does semantically.
I think this version is not even correct though, because the inputs need to be sorted for `std::set_difference`.

Reply:
I intend to revert the commit before merging. Just wanted to show you that the comment is nonsensical.

Reply:
> Just wanted to show you that the comment is nonsensical.

I'm not convinced of that. The loop code is still doing a set difference, and saying so in a comment is very helpful for the reader.

Reply:
> Consider [...] using `std::set_difference` instead of this custom loop

Reply:
`The following loop implements a custom set difference instead of calling the stl implementation because we want to preserve the order as much as possible.`

Or something like that?

Reply:
I pushed something. Please check back.